### ISSUE TITLE
No way to retrieve a line of the validation exception

### ISSUE THREAD

**Issue by wswld** [brunato/xmlschema#61]
Hopefully I'm missing something. But I can't seem to find a way to know on what line of XML input an error occured. It would be very useful in my usecase.

**Comment by brunato** [comment]
The ElementTree library doesn't record XML source line on trees, to do this you need a different parser that tracks lines.

Alternatively to retrieve error source line you can already pass an lxml's tree to the decoder. The tree's elements of this library have an attribute that refers to XML source line: 

```python
>>> import lxml.etree as etree
>>> elem = etree.XML("<A>\n<B1/>\n<B2/>\n</A>")
>>> elem.sourceline
1
>>> elem[0].sourceline
2
>>> elem[1].sourceline
3
```

So passing an lxml tree to the decoder you can already retrieve the line information from `error.elem` attribute.

I could enhance the error message at error class level in case if the target element has a `sourceline` attribute. 
Using a different parser for XML may be evaluated at user level, particularly because the library now uses the *defusedxml* library for protecting from XML attacks.

**Comment by wswld** [comment]
You mean like:

``` py 
xml_schema = XMLSchema(settings.SCHEMA_PATH, validation='lax')
e = etree.XML(request.body)
errors = list(xml_schema.iter_errors(data=e))
```

Or am I missing something because it results in:

```
AttributeError: 'cython_function_or_method' object has no attribute 'replace'
```

Since on some stage comment factory is being passed as a `qname`.

**Comment by brunato** [comment]
A new version is available so you might see the instance line on the displayed validation error message.
Could you please provide a more detailed error message about this `AttributeError` that is generated when you use lxml without removing comments? 

I wish to make lxml's trees usable also when comments are enabled so this could help to fix this bug with instances created with lxml.

Thanks

**Comment by wswld** [comment]
Here's the traceback:

```py
Traceback (most recent call last):
  File "[project_root]/venv/lib/python3.6/site-packages/sanic/app.py", line 471, in handle_request
    response = await response
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/coroutines.py", line 110, in __next__
    return self.gen.send(None)
  File "[project_root]/application/complex/handlers.py", line 84, in post
    errors = list(self.xml_schema_new.iter_errors(data=e))
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/xsdbase.py", line 490, in iter_errors
    for chunk in self.iter_decode(data, path, use_defaults=use_defaults):
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/schema.py", line 662, in iter_decode
    **kwargs):
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/elements.py", line 355, in iter_decode
    for result in type_.iter_decode(elem, validation, **kwargs):
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/complex_types.py", line 461, in iter_decode
    for result in self.content_type.iter_decode(elem, validation, **kwargs):
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/groups.py", line 380, in iter_decode
    for result in xsd_element.iter_decode(child, validation, **kwargs):
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/elements.py", line 355, in iter_decode
    for result in type_.iter_decode(elem, validation, **kwargs):
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/complex_types.py", line 461, in iter_decode
    for result in self.content_type.iter_decode(elem, validation, **kwargs):
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/groups.py", line 370, in iter_decode
    for obj in self.iter_decode_children(elem, index, validation):
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/groups.py", line 524, in iter_decode_children
    for obj in item.iter_decode_children(elem, child_index, validation='lax'):
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/elements.py", line 488, in iter_decode_children
    error = XMLSchemaChildrenValidationError(self, elem, index, self.prefixed_name)
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/validators/exceptions.py", line 114, in __init__
    child_ref = qname_to_prefixed(elem[index].tag, validator.namespaces)
  File "[project_root]/venv/lib/python3.6/site-packages/xmlschema/qnames.py", line 116, in qname_to_prefixed
    return qname.replace(u'{%s}' % uri, '')
AttributeError: 'cython_function_or_method' object has no attribute 'replace'
```

**Comment by brunato** [comment]
Maybe resolved also with _lxml_ data containing comments. Patch available with release 0.9.29.
Bye
